webui/src/pages/auth/users/index.jsx

```diff
@@ -1,51 +1,129 @@
 import useUser from "../../../lib/hooks/user";
 import {ConfirmationButton} from "../../../lib/components/modals";
 import {EntityActionModal} from "../../../lib/components/auth/forms";
 import {Paginator} from "../../../lib/components/pagination";
-import {useRouter} from "../../../lib/hooks/router";
 import {Link} from "../../../lib/components/nav";
 import {
     ActionGroup,
     ActionsBar,
+    AlertError,
     Checkbox,
     DataTable,
-    AlertError,
     FormattedDate,
     Loading,
     RefreshButton
 } from "../../../lib/components/controls";
 import validator from "validator/es";
-import { disallowPercentSign, INVALID_USER_NAME_ERROR_MESSAGE } from "../validation";
-import { resolveUserDisplayName } from "../../../lib/utils";
+import {disallowPercentSign, INVALID_USER_NAME_ERROR_MESSAGE} from "../validation";
+import {resolveUserDisplayName} from "../../../lib/utils";
+import {SearchIcon} from "@primer/octicons-react";
+import InputGroup from "react-bootstrap/InputGroup";
+import {allUsersFromLakeFS} from "../../../lib/components/auth/users";
+import {useRouter} from "../../../lib/hooks/router";
 
+const DEFAULT_DEBOUNCE_MS = 300;
+const DEFAULT_LISTING_AMOUNT = 100;
+const DECIMAL_RADIX = 10;
 const USER_NOT_FOUND = "unknown";
 export const GetUserEmailByIdContext = createContext();
 
 
-const UsersContainer = ({nextPage, refresh, setRefresh, error, loading, userListResults}) => {
+const UsersContainer = ({ refresh, setRefresh, allUsers, loading, error }) => {
     const { user } = useUser();
     const currentUser = user;
 
     const router = useRouter();
-    const after = (router.query.after) ? router.query.after : "";
     const [selected, setSelected] = useState([]);
     const [deleteError, setDeleteError] = useState(null);
     const [showCreate, setShowCreate] = useState(false);
     const [showInvite, setShowInvite] = useState(false);
-    
-    
 
-    useEffect(() => { setSelected([]); }, [refresh, after]);
+    const routerSearchPrefix = (router.query.searchPrefix) ? router.query.searchPrefix : "";
+    const routerUserIdxFromAllUsers = (router.query.userIdxFromAllUsers)
+        ? parseInt(router.query.userIdxFromAllUsers, DECIMAL_RADIX) : 0;
+    const routerHasMorePages = (router.query.hasMorePages) ? router.query.hasMorePages : "false";
+
+    const [searchInput, setSearchInput] = useState(routerSearchPrefix);
+    const [searchPrefix, setSearchPrefix] = useState(routerSearchPrefix);
+    const [prevSearchPrefix, setPrevSearchPrefix] = useState(searchPrefix);
+    const [prevAllUsersLen, setPrevAllUsersLen] = useState(0);
+    const [userIdxFromAllUsers, setUserIdxFromAllUsers] = useState(routerUserIdxFromAllUsers);
+    const [hasMorePages, setHasMorePages] = useState(routerHasMorePages);
+    const [filteredUsers, setFilteredUsers] = useState([]);
+
+
+    useEffect(() => { setSelected([]); }, [refresh]);
 
     const authCapabilities = useAPI(() => auth.getAuthCapabilities());
+
+    useEffect(() => {
```

Comment:
This might work, but it's over-complicating things.
You can simplify this by using the pattern that's used for the Groups search (in your other PR),
By using `useDebouncedState`.
This way, you won't need so many state variables (`searchInput`, `searchPrefix`, `prevSearchPrefix`), and can only use the `routerSearchPrefix`.